content/en/docs/Reference/dependency-resolution.md

```diff
@@ -0,0 +1,351 @@
+
+#### Package constraint
+
+Within a namespace, no two operators may come from the same package.
+
+
+#### Operator Minimization
+
+In the case that there is more than one valid solution, the solution that installs the fewest operators is preferred.
+
+## Best Practices
+
+### Depend on APIs
+
+Operators may add or remove apis at any time - always specify an `olm.gvk` dependency on any APIs your operator requires.
+
+### Set a minimum version
+
+The kubernetes [documentation on api changes](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md#readme) describes what changes are allowed for kube-style operator APIs. These versioning conventions allow an operator to update an API without bumping the `apiVersion`, as long as the api is backwards-compatible.
+
+For operator dependencies, this means that knowing the `apiVersion` of a dependency may not be enough to ensure the dependent operator works as intended. 
+
+For example:
+
+ - `FooOperator v1.0` provides `v1alpha1` of `Foo`
+ - `FooOperator v1.0.1` adds a new field `spec.bar` to `Foo`, but still at `v1alpha1`
+
+Your operator may require the ability to write `spec.bar` into `Foo`. An `olm.gvk` constraint alone is not enough for OLM to determine that you need `FooOperator v1.0.1` and not `FooOperator v1.0`.
+
+Whenever possible, if a specific operator that provides an API is known ahead of time, specify an additional `olm.package` constraint to set a minimum.
+
+### Omit a maximum version, or allow a very wide range
+
+Because operators provide cluster-scoped resources (ApiServices, CRDs), an operator that specifies a small window for a dependency may unnecessarily constrain updates for other consumers of that dependency.
+
+A maximum version should not be set whenever possible, or should be set to a very wide semantic range (i.e. `>1.0.0 <2.0.0`) to prevent conflicts with other operators.
+
+Unlike with conventional package managers, operator authors explicitly encode that updates are safe via OLM's channels. If an update is available for an existing subscription, it comes with the operator provider's promise that it can update from the previous version. Setting a maximum version for a dependency overrides the dependency author's update stream by unnecessarily truncating it at a particular upper bound.
+
+Maximum versions can and should be set, however, if there are known incompatibilties that must be avoided. Note that specific versions can be omitted with the version range syntax, e.g. `> 1.0.0 !1.2.1` 
+
+## Caveats
+
+These are some things to be cautious of when specifying dependencies.
+
+### No compound constraints ("AND")
+
+There is not a way to specify an "AND" relationship between constraints. In other words, there is no way to say: "this operator depends on another operator that both provides `Foo` api and has version `>1.1.0`".
```

Comment:
Ah, this is a pretty big caveat. Up until now I thought the constraints were exactly `olm.package` AND `olm.gvk`.
But now I guess I can see there's nothing to intersect the two constraints on.

Reply:
It might help to have an example that pulls in two operator dependencies, so it's more clear?

Reply:
+1 
Also, do we want to rephrase the statement `There is not a way to specify an "AND" relationship between constraints` or is it just me?